### PR TITLE
fix(core): make ng-version attribute W3C compliant

### PR DESCRIPTION
**Pull request by smartm0use**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Using `ng-version` attribute make the DOM generated not to pass the W3C validation.
The verification gives the following error: "Attribute ng-version not allowed on element div at this point."

Issue Number: #16283

## What is the new behavior?
Now `ng-version` attribute is compliant with W3C validation.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

If developers is using some tool relying on `ng-version` attribute or to grab the version of Angular from the rendered html it will no longer work by searching for the `ng-version` attribute.

### CODE REVIEW

**Comment by smartm0use** [comment]
I signed it!

**Comment by smartm0use** [comment]
I have added and verified to my GitHub account the email I used for CLA.

**Comment by smartm0use** [comment]
I have replaced GitHub email address with the one I used for CLA.

**Comment by smartm0use** [comment]
I have replaced GitHub email address with the one I used for CLA.

**Comment by smartm0use** [comment]
Please check again...

**Comment by smartm0use** [comment]
Please run the check again...

**Comment by Toxicable** [comment]
This is a breaking change.
If anyone is using some tool to grab the version of Angular from the rendered html it will no longer work by searching for the `ng-version` attribute.

**Comment by smartm0use** [comment]
@Toxicable Since the attribute is `data-ng-version` if anyone search for `ng-version` they can still find it.
Anyway I think it is quite mandatory and more important to be compliant with W3C standard...

**Comment by Toxicable** [comment]
@smartm0use I make a quick example showing that you wont be able to find `data-ng-version` if you're looking for `ng-version`
https://jsfiddle.net/hcyahw3c/2/

**Comment by smartm0use** [comment]
@Toxicable I see, thank you for your contribute. So, since some tools have to make some changes... is this anyway a breaking change for Angular developers?

**Comment by Toxicable** [comment]
While yes it is a breaking change since devs could have tools relying on it being this format, however I am uncertain if it's covered under the policy for breaking changes

**Comment by smartm0use** [comment]
@Toxicable I have edited PR description. Thank you.

**Comment by smartm0use** [comment]
Someone please remove the "cla:no" label to this PR.

**Comment by Toxicable** [comment]
You made a commit from an email that has not signed the cla, you should resolve that and also squash you commits

**Comment by smartm0use** [comment]
Anyone here can figure out what is the problem with my PR?
It seems that Travis CI build fails because of test error in `packages/platform-server/test/integration_spec.ts` files. But those lines doesn't exist in that file.

I mean lines 728, 740 and 749 that are reported in the details page.

**Comment by gkalpak** [comment]
The errors correspond to the files as they currently appear on master. By rebasing your PR on master, you should be able to see and fix the errors.

Not sure what's going on on Travis though. Why are the tests run against master and not against the PR head?

**Comment by smartm0use** [comment]
What is the next step? It is my first PR here... usually how long does it take to have a look at a PR?

File: packages/router/test/integration.spec.ts [hunk 1]
```diff
@@ -191,7 +191,7 @@ describe('Integration', () => {
        class OnPushOutlet {
        }
 
-       @Component({selector: 'need-cd', template: `{{'it works!'}}`})
+       @Component({selector: 'need-cd', template: `{{'It works!'}}`})
```
Comment (Toxicable) [inline packages/router/test/integration.spec.ts]: What does changing the casing of this work have to do with your PR?

File: packages/compiler/test/template_parser/template_parser_spec.ts [hunk 2]
```diff
@@ -1224,7 +1224,7 @@ Reference "#a" is defined several times ("<div #a></div><div [ERROR ->]#a></div>
 
         });
 
-        it('should report duplicate reference names when using mutliple exportAs names', () => {
```
Comment (Toxicable) [inline packages/compiler/test/template_parser/template_parser_spec.ts]: Same here, this is unrelated

File: packages/router/test/integration.spec.ts (hunk 1 above)
Comment (smartm0use) [reply packages/router/test/integration.spec.ts]: Nothing, it was one of my tries to fix my PR... Have I to undo it?

File: packages/compiler/test/template_parser/template_parser_spec.ts (hunk 2 above)
Comment (smartm0use) [reply packages/compiler/test/template_parser/template_parser_spec.ts]: Have I to split this PR in two ones? One for `ng-version` and one for typos?

Comment (trotyl) [reply packages/compiler/test/template_parser/template_parser_spec.ts]: A PR could be hanging for months (or years?), so that others may also fix that typo without respecting this one, after that there would be conflicts against master.

**Comment by smartm0use** [comment]
@Toxicable I fixed again my PR. Thank you for your feedback.

**Comment by smartm0use** [comment]
Hello, anyone here can help me to figure out what is missing to make this PR ok?

**Comment by smartm0use** [comment]
Opened new PR.
